docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
 
-.. note::
-   The ``tox -e build_docs_no_examples`` command will build the documentation without
-   executing the :ref:`example notebooks <example_notebooks>`. It will also
-   pass with warnings.
+You can access the documentation landing page by opening
+:file:`docs/_build/html/index.html` with your browser of choice.
+
+When writing documentation, please make sure to fix any warnings that
+arise. To enforce this, the ``build_docs`` environment is set to fail on
+encountering any warnings via the ``-W`` flag to `sphinx-build`_.
+
+You can shorten the tox_ documentation build by running:
+
+.. code-block:: bash
+
+   tox -e build_docs_no_examples
+
+in order to build the documentation without executing the :ref:`example
+notebooks <example_notebooks>`. This command will pass even if there are
+warnings. A tox_ command may also include the ``--parallel`` flag to
+allow a parallel build.
+
+Documentation tools
+===================
+
+ReStructuredText
+----------------
+
+PlasmaPy's documentation is written using the `reStructuredText (reST)`_
+markup language. reST_ is human readable when viewed within a source
+code file or when printed out using `help`. reST_ also contains markup
+that allows the text to be transformed into `PlasmaPy's documentation`_.
+reST_ files end in :file:`.rst`. Documentation contained within
+:file:`.py` files are in the form of docstrings_, which are written in
+reST_.
+
+ReStructuredText Examples
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we show some examples of commonly used reST_ syntax in
+PlasmaPy.  Please refer to the documentation for Sphinx_ and reST_ for
+a list of available directives and syntax.
+
+This is an example of including headings for the document title,
+sections, subsections, and so on. The lines surrounding each heading are
+the same length as that heading.
+
+.. code-block:: rst
+
+   ==============
+   Document title
+   ==============
+
+   Heading 1
+   =========
+
+   Heading 2
+   ---------
+
+   Heading 3
+   ~~~~~~~~~
+
+We can link to code objects by enclosing them in single back ticks.
+This linking will work for Python_ commands as well as certain packages
+like NumPy_, SciPy_, Astropy_, and pandas_. This linking is described in
+the section on :ref:`external-references`. In-line code examples are
+enclosed in double back ticks.
+
+.. code-block:: rst
+
+   Here is a reference to `plasmapy.particles` that will show the full
+   namespace in the rendered documentation.
+
+   Only the word "Particle" will show up if a prepend a tilde like in
+   `~plasmapy.particles.particle_class.Particle`.
+
+   Here is some in-line code: ``import astropy.units as u``.
+
+This reST_ block will render as:
+
+   Here is a reference to `plasmapy.particles` that will show the full
+   namespace in the rendered documentation.
+
+   Only the word "Particle" will show up if a prepend a tilde like in
+   `~plasmapy.particles.particle_class.Particle`.
+
+   Here is some in-line code: ``import astropy.units as u``.
```

Comment:
```suggestion
.. code-block:: rst

   Here `plasmapy.particles` provides a linked reference to the
   module's documentation.
   
   Adding a tilde at the beginning `~plasmapy.particles` still
   provides a linked reference to the associated documentation
   but shortens the display so only "particles" is displayed.

   Double backticks are used to show inline code that is not
   cross-referenced: ``import astropy.units as u``.

This reST_ block will render as:

   Here `plasmapy.particles` provides a linked reference to the
   module's documentation.
   
   Adding a tilde at the beginning `~plasmapy.particles` still
   provides a linked reference to the associated documentation
   but shortens the display so only "particles" is displayed.

   Double backticks are used to show inline code that is not
   cross-referenced: ``import astropy.units as u``.
```

I felt this could use a little more elaboration for clarity, so this is my take on it.  I mostly wanted to emphasize that a link to the documentation is generated.